internal/storage/storage.go

```diff
@@ -58,42 +73,75 @@ func (s *Storage) Decrypt() error {
 	return s.parse(decodedData)
 }
 
-// Save tries to encrypt and save the storage.
-func (s *Storage) Save() error {
-	plaintext, err := json.Marshal(s.Namespaces)
+// DecryptV2 tries to decrypt the newer age storage.
+func (s *Storage) DecryptV2() error {
+	rawFile, err := os.Open(s.File)
+	if err != nil {
+		return BackendError{Message: err.Error()}
+	}
+	defer rawFile.Close()
+
+	identity, err := age.NewScryptIdentity(s.Password)
+	if err != nil {
+		return BackendError{Message: err.Error()}
+	}
+
+	armorFile := armor.NewReader(rawFile)
+
+	cryptFile, err := age.Decrypt(armorFile, identity)
 	if err != nil {
 		return BackendError{Message: err.Error()}
 	}
 
-	missing := aes.BlockSize - (len(plaintext) % aes.BlockSize)
-	padded := make([]byte, len(plaintext)+missing)
+	decodedData := &bytes.Buffer{}
 
-	copy(padded, plaintext)
+	_, err = io.Copy(decodedData, cryptFile)
+	if err != nil {
+		return BackendError{Message: err.Error()}
+	}
 
-	plaintext = padded
+	return s.parse(decodedData.Bytes())
+}
 
-	if len(plaintext)%aes.BlockSize != 0 {
-		return BackendError{Message: "plaintext is not a multiple of the block size"}
+// Decrypt tries to decrypt the storage.
+func (s *Storage) Decrypt() error {
+	if err := s.DecryptV1(); err == nil {
+		return nil
 	}
 
-	block, err := aes.NewCipher(s.Password)
+	return s.DecryptV2()
+}
+
+// Save tries to encrypt and save the storage.
+func (s *Storage) Save() error {
+	rawFile, err := os.Create(s.File)
```

Comment:
This is pretty unsafe.

I should write to a tempfile first and then rename it.

Reply:
Fixed, although I have no idea if that tmpfile/rename pattern works on windows.